DataStructure/src/main/list/ResizableArray.java

```diff
@@ -35,14 +43,19 @@ public void ensureCapacity(){
             list = list2;
         }
     }
+
     /**
      * Adds an element to the array.
      * @param num
      */
-    public void add(int num){
-        ensureCapacity();
-        list[writeIndex] = num;
-        writeIndex++;
+    public boolean add(int num){
+            ensureCapacity();
+            list[writeIndex] = num;
+            writeIndex++;
+            if (list[writeIndex - 1] == num){
```

Comment:
This if statement is not needed because it will never result false, it will throw exception if the writeIndex reaches to Max int value